src/util/saveArticles.ts

```diff
@@ -402,9 +403,32 @@ async function treatVideo(mw: MediaWiki, dump: Dump, srcCache: KVS<boolean>, art
     }
 
     sourceEl.setAttribute('src', newUrl);
+
+    /* Scrape subtitle */
+    const trackEle = videoEl.querySelector('track');
+    if (trackEle) {
+        const { content, trackTitle } = await treatSubtitles(trackEle, downloader, webUrlHost, mw);
+        const article = new ZimArticle({ url: `${trackTitle}.vtt`, mimeType: 'text/vtt', data: content, ns: 'I' });
+        zimCreator.addArticle(article);
+        trackEle.setAttribute('src', `${getRelativeFilePath(articleId, trackTitle, 'I')}.vtt`);
+    }
     return { mediaDependencies };
 }
 
+export async function treatSubtitles(trackEle: any, downloader: Downloader, webUrlHost: string, mw: MediaWiki) {
+    try {
+        const sourceUrl = getFullUrl(webUrlHost, trackEle.getAttribute('src'), mw.base);
+        const trackformat: any = QueryStringParser.parse(sourceUrl).trackformat;
```

Comment:
Yeah even I was confused to get this weird casting error, if I put type string:

`Type 'ParsedUrlQuery' is not assignable to type 'string'`, so the quick fix was to change to any type. Can you suggest how to fix it?
<img width="682" alt="Screenshot 2020-05-17 at 1 37 40 AM" src="https://user-images.githubusercontent.com/13726478/82136787-1c81ef00-97df-11ea-9a1d-d3f9e7667b92.png">

Reply:
@bakshiutkarsha try this:
![image](https://user-images.githubusercontent.com/6795530/82139715-a4451900-9832-11ea-818c-7506b5cc4e94.png)
it's not required to specify the types of these two properties because they follows from `parse()` definition. However, `trackformat` defined as `string | string[]` there, so that we have to add explicit type guard bellow while calling `replace()`.
Extra point: please follow common variable naming style (`trackformat` -> `trackFormat`) as well ;)

Reply:
UPD: same thing for `trackTitle` bellow ;)
Another approach would be to specify the types after all, but it wouldn't look decent IMO:
```
const {trackTitle, trackformat} = QueryStringParser.parse(sourceUrl) as {trackTitle: string, trackformat: string};
```

Reply:
@midik Thanks a lot for the help, I used the latter solution, because it only requires to define the type once, which is more efficient.

Also, for following common variable naming style, after doing the props destructing changes, I am getting the "trackformat" from url directly, so can't do much there.